src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -191,224 +204,233 @@ private void CreateStreamAndStartWatcher()
                 _watcherThread.Start();
                 _runLoopStartedEvent.WaitOne();
             }
-        }
 
-        private void WatchForFileSystemEventsThreadStart()
-        {
-            // Get this thread's RunLoop
-            _watcherRunLoop = Interop.RunLoop.CFRunLoopGetCurrent();
-            Debug.Assert(_watcherRunLoop != IntPtr.Zero);
-
-            // Schedule the EventStream to run on the thread's RunLoop
-            Interop.EventStream.FSEventStreamScheduleWithRunLoop(_eventStream, _watcherRunLoop, Interop.RunLoop.kCFRunLoopDefaultMode);
-            if (Interop.EventStream.FSEventStreamStart(_eventStream))
+            private void WatchForFileSystemEventsThreadStart()
             {
-                // Notify the StartRaisingEvents call that we are initialized and about to start
-                // so that it can return and avoid a race-condition around multiple threads calling Stop and Start
-                _runLoopStartedEvent.Set();
+                // Get this thread's RunLoop
+                _watcherRunLoop = Interop.RunLoop.CFRunLoopGetCurrent();
+                Debug.Assert(_watcherRunLoop != IntPtr.Zero);
 
-                // Start the OS X RunLoop (a blocking call) that will pump file system changes into the callback function
-                Interop.RunLoop.CFRunLoopRun();
+                // Schedule the EventStream to run on the thread's RunLoop
+                Interop.EventStream.FSEventStreamScheduleWithRunLoop(_eventStream, _watcherRunLoop, Interop.RunLoop.kCFRunLoopDefaultMode);
+                if (Interop.EventStream.FSEventStreamStart(_eventStream))
+                {
+                    // Notify the StartRaisingEvents call that we are initialized and about to start
+                    // so that it can return and avoid a race-condition around multiple threads calling Stop and Start
+                    _runLoopStartedEvent.Set();
```

Comment:
This Set is the first thing that happen in both the if and the else.  Can it be moved out? e.g.
```C#
bool successfullyStarted = Interop.EventStream.FSEventStreamStart(_eventStream);
_runLoopStartedEvent.Set();
if (successfullyStarted)
{
    ...
}
else
{
    ...
}
```